apps/setting/models_provider/impl/azure_model_provider/azure_model_provider.py

```diff
@@ -11,18 +11,39 @@
 from common.util.file_util import get_file_content
 from setting.models_provider.base_model_provider import IModelProvider, ModelProvideInfo, ModelInfo, \
     ModelTypeConst, ModelInfoManage
+from setting.models_provider.impl.azure_model_provider.credential.embedding import AzureOpenAIEmbeddingCredential
 from setting.models_provider.impl.azure_model_provider.credential.llm import AzureLLMModelCredential
 from setting.models_provider.impl.azure_model_provider.model.azure_chat_model import AzureChatModel
+from setting.models_provider.impl.azure_model_provider.model.embedding import AzureOpenAIEmbeddingModel
 from smartdoc.conf import PROJECT_DIR
 
 base_azure_llm_model_credential = AzureLLMModelCredential()
+base_azure_embedding_model_credential = AzureOpenAIEmbeddingCredential()
 
 default_model_info = ModelInfo('Azure OpenAI', '具体的基础模型由部署名决定', ModelTypeConst.LLM,
                                base_azure_llm_model_credential, AzureChatModel, api_version='2024-02-15-preview'
                                )
 
-model_info_manage = ModelInfoManage.builder().append_default_model_info(default_model_info).append_model_info(
-    default_model_info).build()
+embedding_model_info = [
+    ModelInfo('text-embedding-3-large', '具体的基础模型由部署名决定', ModelTypeConst.EMBEDDING,
+              base_azure_embedding_model_credential, AzureOpenAIEmbeddingModel, api_version='2023-05-15'
+              ),
+    ModelInfo('text-embedding-3-small', '', ModelTypeConst.EMBEDDING,
+              base_azure_embedding_model_credential, AzureOpenAIEmbeddingModel, api_version='2023-05-15'
+              ),
+    ModelInfo('text-embedding-ada-002', '', ModelTypeConst.EMBEDDING,
+              base_azure_embedding_model_credential, AzureOpenAIEmbeddingModel, api_version='2023-05-15'
+              ),
+]
+
+model_info_manage = (
+    ModelInfoManage.builder()
+    .append_default_model_info(default_model_info)
+    .append_model_info(default_model_info)
+    .append_model_info_list(embedding_model_info)
+    .append_default_model_info(embedding_model_info[0])
+    .build()
+)
 
 
 class AzureModelProvider(IModelProvider):
```

Comment:
The provided code snippet has a few issues that need to be addressed:

1. **Duplicate Default Model Information**: In `ModelInfoManage`, there are two duplicate calls to `.append_default_model_info`. This can lead to unexpected behavior, as models might only appear once in the final list.

2. **Incorrect Credential Type**: The usage of `AzureOpenAIEmbeddingCredential` seems incorrect. While `AzureOpenAIEmbeddingCredential` is likely meant to handle embedding credentials, it may not be compatible with other components that expect different credential types (`BaseCredentialProvider`). You should ensure consistency on what type of credential is used throughout your application.

3. **Potential Naming Conflicts**: Ensure all identifiers (classes, variables) have consistent naming conventions and do not conflict with existing Python built-in functions or keywords.

4. **Improper List Concatenation**: When creating `embedding_model_info_list`, you should directly pass the list of `ModelInfo`, rather than trying to build another instance of `ModelInfoManage`.

Here's a revised version of the code addressing these issues:

```python
import logging

from common.util.file_util import get_file_content
from setting.models_provider.base_model_provider import IModelProvider, ModelProvideInfo, ModelInfo, \
    ModelTypeConst, ModelInfoManage
from setting.models_provider.impl.azure_model_provider.credential.embedding import AzureOpenAIEmbeddingCredential
from setting.models_provider.impl.azure_model_provider.credential.llm import AzureLLMModelCredential
from setting.models_provider.impl.azure_model_provider.model.azure_chat_model import AzureChatModel
+from setting.models_provider.impl.azure_model_provider.model.embedding import AzureOpenAIEmbeddingModel
from smartdoc.conf import PROJECT_DIR

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

default_model_info = ModelInfo(
    name='Azure OpenAI',
    description='具体的基础模型由部署名决定',
    model_type=ModelTypeConst.LLM,
    model_credential=AzureLLMModelCredential(),
    provider_class=AzureChatModel,
    api_version='2024-02-15-preview'
)

# Assuming you want to reuse default_model_info in some context later
reuse_default_model_info = False

if not reuse_default_model_info:
    logger.info("Appending default model info.")
else:
    logger.warning("Default model info will not be repeated.")

azure_embedding_model_credentials = [
    AzureOpenAIEmbeddingCredential(sdk_key="your-sdk-key"),
]

# Create AzureEmbeddingModel instances for each entry in azure_embedding_model_credentials
azure_embedding_models = [AzureOpenAIEmbeddingModel(api_version=cred.api_version) for cred in azure_embedding_model_credentials]

embedding_model_infos = [
    ModelInfo(
        name=model_name,
        description=None,  # Set if needed
        model_type=ModelTypeConst.EMBEDDING,
        model_credential=next(filter(lambda c: isinstance(c, AzureOpenAIEmbeddingCredential), model_infos))
    ) for model_name, _, _, model_infos in zip(
        ['text-embedding-3-large', 'text-embedding-3-small', 'text-embedding-ada-002'],
        [(None,) * 3],  # Placeholder values
        [True] * 3,      # Assume True for demonstration purposes
        [(AzureOpenAIEmbeddingCredential(sdk_key=f"{i}-key") for i in range(3))]   # Dynamically generate AWSIAMCredentials
    )
]

model_info_manage = (
    ModelInfoManage.builder()
    .append_default_model_info(default_model_info)
    .append_model_info(reuse_default_model_info and embedding_model_infos[0])  # Use the first embedded model info if reuse is desired
    .append_model_info(embedding_model_info[-1])  # Append the last embedded model info
    .build()
)
```

### Key Changes:
- Removed unnecessary duplication of default model information.
- Corrected the credential type in `AzureOpenAIEmbeddingCredential`.
- Used proper object creation patterns when defining embedding models.
- Ensured list concatenations are done correctly using simple list assignments.